src/Microsoft.Data/BaseColumn.cs

```diff
@@ -59,16 +59,25 @@ protected set
             get => GetValue(startIndex, length);
         }
 
+        /// <summary>
+        /// Called internally from Merge and GroupBy
+        /// </summary>
+        /// <param name="length"></param>
+        public virtual void Resize(long length) => throw new NotImplementedException();
+
         /// <summary>
         /// Clone column to produce a copy potentially changing the order by supplying mapIndices and an invert flag
         /// </summary>
         /// <param name="mapIndices"></param>
         /// <param name="invertMapIndices"></param>
         /// <returns></returns>
         public virtual BaseColumn Clone(BaseColumn mapIndices = null, bool invertMapIndices = false, long numberOfNullsToAppend = 0) => throw new NotImplementedException();
+        public virtual BaseColumn Clone(IEnumerable<long> mapIndices, long numberOfNullsToAppend = 0) => throw new NotImplementedException();
```

Comment:
Do we need 2 `virtual` Clone methods? Is one a more general version of the other? It would be nice if we could figure out a way to only have a single `virtual Clone` method that takes the appropriate parameters.